python-frontend/src/test/java/org/sonar/python/lexer/PythonLexerTest.java

```diff
@@ -274,6 +274,20 @@ void fstring_triple_double_quote() {
       hasToken("\"\"\"", PythonTokenType.FSTRING_END)));
   }
 
+  // Lambdas and walrus operators should be surrounded by parenthesis in an FString
```

Comment:
_Nitpick:_ It's nice to have one case of lambda and walrus as well as one case that is parsed as the "expected" syntax and one that is actually parsed as a format specifier. Because we have 2 tests that change both variables at once, it's not obvious to see the relationship though.

I think it would be nice to have one additional test for a lambda that is correctly parsed as a lambda (or a walrus that is lexed as a format specifier). No need to have all combinations of course, but I feel one more could help with understanding what's going on in a clear way.

Reply:
I see what you mean. Let's keep it so we don't have to re-escape it after